sui-execution/latest/sui-adapter/src/programmable_transactions/context.rs

```diff
@@ -563,20 +578,22 @@ impl<'vm, 'state, 'a> ExecutionContext<'vm, 'state, 'a> {
                 inner: ResultValue { value, .. },
             } = input;
             let Some(object_metadata) = object_metadata_opt else { return Ok(()) };
-            let is_mutable_input = object_metadata.is_mutable_input;
-            let owner = object_metadata.owner;
-            let id = object_metadata.id;
-            input_object_metadata.insert(object_metadata.id, object_metadata);
+            let id = object_metadata.id();
+            input_object_metadata.insert(id, object_metadata.clone());
+            // We don't count receiving object arguments as something that needs to be written
+            // unless it has actually been received -- this is tracked in the `received` field of
+            // the object store.
+            if let InputObjectMetadata::Receiving { .. } = &object_metadata { return Ok(()) }
```

Comment:
Shouldn't we early return above if we are in this case? Then we shouldn't need to clone.

Reply:
(you might need to copy out the owner/id fields again but should be easy enough